app/models/assignment.js

```diff
@@ -34,6 +34,47 @@ export default class AssignmentModel extends Model {
   // see the my projects controllers
   sortingActualEndDate;
 
+  // if the each dcpPublichearinglocation and dcpDateofpublichearing properties are filled in dispositions array,
+  // then hearings have been submitted for that project
+  @computed('dispositions.@each.{dcpPublichearinglocation,dcpDateofpublichearing}')
+  get hearingsSubmitted() {
+    const dispositions = this.get('dispositions');
+    // array of hearing locations
+    const dispositionHearingLocations = dispositions.map(disp => `${disp.dcpPublichearinglocation}`);
+    // array of hearing dates
+    const dispositionHearingDates = dispositions.map(disp => disp.dcpDateofpublichearing);
+    // hearingsSubmittedForProject checks whether each item in array is truthy
+    const hearingsSubmittedForProject = dispositionHearingLocations.every(location => !!location) && dispositionHearingDates.every(date => !!date);
+    return hearingsSubmittedForProject;
+  }
+
+  // if all dcpPublichearinglocation in dispositions array equal "waived",
+  // then hearings have been waived
+  @computed('dispositions.@each.dcpPublichearinglocation')
+  get hearingsWaived() {
+    const dispositions = this.get('dispositions');
+
+    // array of hearing locations
+    const dispositionHearingLocations = dispositions.map(disp => disp.dcpPublichearinglocation);
+    // each location field equal to 'waived'
+
+    return dispositionHearingLocations.every(location => location === 'waived');
+  }
+
+  @computed('hearingsSubmitted', 'hearingsWaived')
+  get hearingsSubmittedOrWaived() {
+    const hearingsSubmitted = this.get('hearingsSubmitted');
+    const hearingsWaived = this.get('hearingsWaived');
+    return !!hearingsSubmitted || !!hearingsWaived;
+  }
```

Comment:
We know that both hearingsSubmitted and hearingsWaived return booleans, so no need to coerce to boolean with the double bang. I would just simplify:

```
return this.hearingsSubmitted || this.hearingsWaived
```

It's shorter, but there's also much less noise.

Reply:
Will do